python/ray/_private/ray_microbenchmark_helpers.py

```diff
@@ -14,7 +14,7 @@
 def timeit(
     name, fn, multiplier=1, warmup_time_sec=10
 ) -> List[Optional[Tuple[str, float, float]]]:
-    if filter_pattern not in name:
+    if filter_pattern and name not in filter_pattern:
```

Comment:
could this cause weird matching behavior?

e.g. filter = "A AB C"
e.g. name = "B"

Reply:
yeah - this is not perfect. 
The assumption is that we will less likely have microbenchmarks with overlapping names. (`AB` and `B` will not co-exist) 
So one should just provide the full names in the string to specify multiple tests to be run. 

I feel this is more useful than the previous matching patten since the previous is used to run a subset of test with common strings in name: 
```
filter = "A"
Tests = ["AB", "AC", "AD", ..] 
```

But there is no way to say run `BB` and `AA` together.

Reply:
should we document the behavior a little bit more ? seems like it could match substrings to the filter